app/controllers/BasicAuthenticationFilter.scala

```diff
@@ -112,33 +213,62 @@ object BasicAuthenticationFilterConfiguration {
       }
 
     val enabled = boolean("enabled").getOrElse(false)
-
-    val credentials: Option[(String, Set[String])] = for {
-      username <- string("username")
-      passwords <- seq("password")
-    } yield (username, passwords.toSet)
-
-    val (username, passwords) = {
-      def uuid = UUID.randomUUID.toString
-      credentials.getOrElse((uuid, Set(uuid)))
-    }
-
-    def realm(hasCredentials: Boolean) = {
-      val realm = string("realm").getOrElse(defaultRealm)
-      if (hasCredentials) realm
-      else credentialsMissingRealm(realm)
-    }
+    val ldapEnabled = boolean("ldap.enabled").getOrElse(false)
 
     val excluded = configuration.getStringSeq(root + "excluded")
       .getOrElse(Seq.empty)
       .toSet
 
-    BasicAuthenticationFilterConfiguration(
-      realm(credentials.isDefined),
-      enabled,
-      username,
-      passwords,
-      excluded
-    )
+    if (ldapEnabled) {
+      val connection: Option[(String, Int)] = for {
+        server <- string("ldap.server")
+        port <- int("ldap.port")
+      } yield (server, port)
+
+      val (server, port) = {
+        connection.getOrElse(("localhost", 389))
+      }
+
+      val username = string("ldap.username").getOrElse("")
+      val password = string("ldap.password").getOrElse("")
+
+      val searchDN = string("ldap.search-base-dn").getOrElse("")
+      val searchFilter = string("ldap.search-filter").getOrElse("")
+      val connectionPoolSize = int("ldap.connection-pool-size").getOrElse(10)
+      val sslEnabled = boolean("ldap.ssl").getOrElse(false)
+
+      BasicAuthenticationFilterConfiguration(
+        string("realm").getOrElse(defaultRealm),
+        enabled,
```

Comment:
@patelh @dpavlov  is this a bug, did you mean `ldapEnabled`? Or are we supposed to also set `basicAuthentication.enabled=true` to have LDAP?  Maybe the documentation could use some clarification here.

Reply:
Yes you are right. LDAP works only with basic auth. Docs also have been changed to make it clear.